cmip6_downscaling/workflows/utils.py

```diff
@@ -39,42 +96,92 @@ def delete_chunks_encoding(ds):
             del ds[coord].encoding['chunks']
 
 
-def rechunk_zarr_array(zarr_array, chunks_dict, connection_string, max_mem="10MB"):
-    """[summary]
+def make_rechunker_stores(
+    connection_string: str,
+):  # -> tuple[fsspec.mapping.FSmap, fsspec.mapping.FSmap]:
+    """Initialize two stores for rechunker to use as temporary and final rechunked locations
 
     Parameters
     ----------
-    ds : [zarr array]
-        [description]
-    chunks_dict : dict
-        Desired chunks sizes for each variable. They can either be specified in tuple or dict form.
-        But dict is probably safer! When working in space you proabably want somehting like
-        (1, -1, -1) where dims are of form (time, lat, lon). In time you probably want
-        (-1, 10, 10). You likely want the same chunk sizes for each variable.
     connection_string : str
-        [description]
+        Connection string to give you write access to the stores
+
+    Returns
+    -------
+    temp_store, target_store, path_tgt : tuple[fsspec.mapping.FSmap, fsspec.mapping.FSmap, string]
+        Stores where rechunker will write and the path to the target store
+    """
+    path_tmp = "az://cmip6/temp/{}.zarr".format(temp_file_name())
+    path_tgt = "az://cmip6/temp/{}.zarr".format(temp_file_name())
+    temp_store = fsspec.get_mapper(path_tmp, connection_string=connection_string)
+    target_store = fsspec.get_mapper(path_tgt, connection_string=connection_string)
+    return temp_store, target_store, path_tgt
+
+
+def rechunk_zarr_array(
```

Comment:
wish list for this function :) 
1. make it work for all variables in the ds, 
2. make it "lazy" so one can call it just in case without incurring unnecessary overhead.

Reply:
it is lazy now!

Reply:
Passing (1) to an issue.